config/compilerEnv.cmake

```diff
@@ -68,134 +68,138 @@ endmacro()
-            "${PROJECT_SOURCE_DIR}/config/capture_lcov.sh and "
-            "${DRACO_DIR}/cmake/capture_lcov.sh" )
-        endif()
-        add_custom_command(
-          OUTPUT "${PROJECT_BINARY_DIR}/covrep_target_aways_out_of_date.txt"
-          BYPRODUCTS
-            "${PROJECT_BINARY_DIR}/coverage.info"
-            "${PROJECT_BINARY_DIR}/coverage.txt"
-          COMMAND ${LCOV} ${lcovopts2} --capture --directory .
-          COMMAND ${LCOV} ${lcovopts2} --remove coverage.info ${lcov_ignore}
-          COMMAND genhtml coverage.info --demangle-cpp --output-directory cov-html
-          # COMMAND ${LCOV} ${lcovopts1} --list coverage.info
-          COMMAND "${captureLcov}" -g "${GCOV}" -l "${LCOV}"
-          )
-        unset( captureLcov )
-        add_custom_target( covrep
-          DEPENDS "${PROJECT_BINARY_DIR}/covrep_target_aways_out_of_date.txt"
-          COMMENT "
-==> View text coverage report in file coverage.txt.
-==> View HTML coverage report with command: firefox cov-html/index.html
-==> Repeat text coverage report with command: lcov --list coverage.info
-" )
-        message( STATUS "Code coverage build ... enabled ('make covrep' to "
-          "see a text and/or a html report)")
-        message("CODE_COVERAGE_IGNORE_REGEX  = ${CODE_COVERAGE_IGNORE_REGEX}")
-        block_indent( 90 27
-          "CODE_COVERAGE_IGNORE_REGEX = ${CODE_COVERAGE_IGNORE_REGEX}")
-      else()
-        message( STATUS "Code coverage build ... disabled (lcov and/or gcov "
-          "not found)" )
-      endif()
+    # shared or static libraries?
+    if( ${DRACO_LIBRARY_TYPE} MATCHES "STATIC" )
+      set( DRACO_SHARED_LIBS 0 )
+    elseif( ${DRACO_LIBRARY_TYPE} MATCHES "SHARED" )
+      # This CPP symbol is used by config.h to signal if we are need to add
+      # declspec(dllimport) or declspec(dllexport) for MSVC.
+      set( DRACO_SHARED_LIBS 1 )
+      mark_as_advanced(DRACO_SHARED_LIBS)
     else()
-      message( STATUS "Code coverage build ... disabled (CMAKE_BUILD_TYPE "
-        "!= Debug" )
+      message( FATAL_ERROR "DRACO_LIBRARY_TYPE must be set to either STATIC or "
+      "SHARED.")
     endif()
-  endif()
+    set( DRACO_SHARED_LIBS "${DRACO_SHARED_LIBS}" CACHE BOOL
+      "This CPP symbol is used by config.h to signal if we are need to add declspec(dllimport) or declspec(dllexport) for MSVC." )
+
+    #--------------------------------------------------------------------------#
+    # Setup common options for targets
+    #--------------------------------------------------------------------------#
+
+    # Control the use of interprocedural optimization. This used to be set by
+    # editing compiler flags directly, but now that CMake has a universal
+    # toggle, we use it. This value is used in component_macros.cmake when
+    # properties are assigned to individual targets.
+
+    #  See https://cmake.org/cmake/help/git-stage/policy/CMP0069.html
+    if( WIN32 )
+      set( USE_IPO OFF CACHE BOOL
```

Comment:
Is IPO not available at all for Win32 builds, or do we just not support it?

Reply:
We don't support it because it causes some build issues and (currently) we don't really care if mcgrid is 10% faster.  :smile: